### ISSUE TITLE
Remove input file parameters via CLI arguments

### ISSUE THREAD

**Issue by joshuahansel** [idaholab/moose#20549]
## Reason
There are cases when a single input file can be used for multiple tests. When we want to modify an existing parameter or add a new one, we can do this via command-line arguments. However, there is nothing that can be done when a parameter needs to be removed from the input file, so we are forced to duplicate the input file (at least part of it) in this situation.

## Design
This would add some ability to remove parameters from an input file via command-line arguments. For example, this might be the syntax to remove a parameter `Functions/myfunc/paramC`:
```
someapp-opt -i inputfile.i -Functions/myfunc/paramC
```
I don't know enough about CLI parsing to know whether `-` is treated specially, so it may not be a good example.

## Impact
This will add a new capability.

**Comment by joshuahansel** [comment]
@GiudGiud

**Comment by GiudGiud** [comment]
I'd be worried about how -Functions would get parsed to -F flag 
Maybe --remove?
@dschwen any insight given your work on HIT syntax ?

**Comment by permcody** [comment]
Yeah, I wouldn't recommend using a dash to mean something other than a flag. I do think we could make something like this work though. It's not really a HIT thing.

**Comment by YaqiWang** [comment]
How about `Functions/myfunc/paramC=` syntax? Empty means delete basically.

**Comment by GiudGiud** [comment]
Seems shorter but maybe less intuitive?

**Comment by permcody** [comment]
The problem with an empty left hand side operand is that if we put this on the command-line and somebody adds more options after the equals, it starts to look really screwy:
`Functions/myfunc/paramC= Functions/myotherfunc/override_param=new_value`

Looks a lot like the valid C++ expression:
`A = B = C`

**Comment by joshuahansel** [comment]
Yeah, I personally prefer `--remove` because it's more explicit. Also, it seems like it could get confused with `param=''`.

@GiudGiud What do you mean by "on my radar"? As in, do I plan on doing this right now? I could do it if it's simple, but I might need someone to point me in the right direction.

**Comment by GiudGiud** [comment]
I think --remove is the right direction. Just pinged @dschwen

**Comment by dschwen** [comment]
Yeah, fine with me although I would also like the `param= ` syntax

**Comment by GiudGiud** [comment]
param= syntax should not work in the input file btw.
As in it should still error when this pattern of an empty equal is found.
This might be a detail but it will be an inconsistency between cli-args and input files

**Comment by YaqiWang** [comment]
Actually, the input syntax with `param =` can be considered as unprovided. In trainings, we can use it as place holders. Then things will be consistent. Even not doing this, the parser can possibly make difference between command line and input file. This syntax is very easy to use. I do not have strong opinions on this though.

**Comment by GiudGiud** [comment]
Unprovided errors out correctly right now, it's just that it looks like the "unprovided" syntax. 
The parser can make a difference between CLI and input file.

**Comment by joshuahansel** [comment]
I just tried out situations without any code changes:

1. `param=` in the input file:
   ```
   *** ERROR ***
   /Users/hansje/projects/sockeye/test/tests/components/heat_pipe_vo/heat_pipe_vo.i:128: invalid character '=' - did you 
   leave a field value blank after a previous '='?
   ```
   Here, the missing value is on line 127, not 128.

2. Overwriting a set `Real` parameter in input file with `param=` in CLI args:
   ```
   *** ERROR ***
   /Users/hansje/projects/sockeye/test/tests/components/heat_pipe_vo/heat_pipe_vo.i:106: invalid float syntax for 
   parameter: Components/heat_pipe/D_wick_i=
   ```
   The parameter was defined on line 106 (but the fact it mentions a line at all is strange because it was overwritten on the command line).
3. Overwriting a set `std::string` parameter in input file with `param=` in CLI args: no error, just interprets it as `param=''`.

I think example 3 is a definite problem for `param=`.

If there are no objections, I suggest we start with `--remove`. If someone feels passionately about `param=` down the line, that can be done additionally.

**Comment by GiudGiud** [comment]
Remind me to go fix all the test inputs in the reactor module so that they run standalone once we have this.

Missing this capability is the reason why cli_args are getting big in their tests, and why some of the files dont run without the cli_args

**Comment by YaqiWang** [comment]
This can be useful when a parameter is included in `[GlobalParams]` but one object does not want the parameter being set. In such a case, users can add `TheObject/unrequired_param=` to unset the parameter and make the parameter at invalid state.

**Comment by joshuahansel** [comment]
I'd like to do something on this if it's not beyond me and not too time-consuming. Does someone know the way and can point me in the right direction? (Managing CLI flags and messing around with the input tree after parsing the input file)

**Comment by GiudGiud** [comment]
no idea.
@dschwen do you think a new Walker would be a good design for this?

Or should Josh just add new code somewhere in Parser::parse to remove the params from what got parsed

**Comment by permcody** [comment]
I added additional keywords to the title because I can never seem to find this issue when it's on my mind. I finally ran into a case where I could have used this and no, the obvious workarounds were not ideal so I ended up duplicating the file and removing the one argument...

A couple of thoughts. First we already have the `inactive` keyword, which is valid in every block for turning off sub-blocks. We could simply extend the use of that keyword to apply to blocks as well. I would expect this to be a relatively easy addition.

I haven't really dived into all of Robert's logic there in the past because it has been solid. I'm not sure if building a new walker is the right idea or not because we must process the CLI options for the final parse tree. 

Anyway, Josh, let us know if you are up for this otherwise, I'll see if I can find someone here to work on this. I would expect all or nearly all changes to occure in `Parser.C`

**Comment by joshuahansel** [comment]
@permcody I'm completely swamped at the moment, so this isn't currently on my radar.

The `inactive` idea may be good enough (even as is, for just sub-blocks only). I have forgotten about it sometimes because it's often unattractive (as unattractive as just omitting the sub-block from the input file and putting it only in CLI args):
- Need to repeat all of the parameters of the disabled sub-block in the `tests` (which can be a large list for some objects like components)
- Assuming object `blah_old` has been made inactive and replaced by `blah_new`, any references to the old object name in the input file also need to be changed.

**Comment by GiudGiud** [comment]
the other issue is that inactive cannot be specified at the same time as active. So that conflicts in some inputs
